app/frontends/capabilities.ts

```diff
@@ -0,0 +1,141 @@
+import HarmonyRequest from '../models/harmony-request';
+import { Response, NextFunction } from 'express';
+import { keysToLowerCase } from '../util/object';
+import { RequestValidationError } from '../util/errors';
+import { CmrCollection, getCollectionsByIds, getCollectionsByShortName, getVariablesForCollection } from '../util/cmr';
+import { addCollectionsToServicesByAssociation } from '../middleware/service-selection';
+import _ from 'lodash';
+import { ServiceConfig } from '../models/services/base-service';
+import { harmonyCollections } from '../models/services';
+
+interface CollectionCapabilities {
+  conceptId: string;
+  shortName: string;
+  variableSubset: boolean;
+  bboxSubset: boolean;
+  shapeSubset: boolean;
+  concatenate: boolean;
+  reproject: boolean;
+  outputFormats: string[];
+  services: ServiceConfig<unknown>[];
+  variables: string[];
+}
+
+/**
+ * Loads the collection info from CMR using short name or concept ID passed into the harmony
+ * request
+ *
+ * @param req - The request sent by the client
+ * @returns the collection info
+ * @throws RequestValidationError if no collection is found or parameters are invalid
+ */
+async function loadCollectionInfo(req: HarmonyRequest): Promise<CmrCollection> {
+  const query = keysToLowerCase(req.query);
+  const { collectionid, shortname } = query;
+  let collections;
+  let pickedCollection;
+  if (!collectionid && !shortname) {
+    throw new RequestValidationError('Must specify either collectionId or shortName');
+  } else if (collectionid && shortname) {
```

Comment:
nit: these don't really need to be `else'd` since you are throwing in each case.

Reply:
While redundant I prefer this style - IMO makes it more obvious when branches are taken vs not.

Reply:
nit: I would prefer refactor the validation code into its only function to make this large function more readable.